Common/Securities/Option/OptionFilterUniverse.cs

```diff
@@ -224,6 +225,483 @@ public OptionFilterUniverse PutsOnly()
         {
             return Contracts(contracts => contracts.Where(x => x.ID.OptionRight == OptionRight.Put));
         }
+
+        /// <summary>
+        /// Sets universe of a single call contract with the closest match to criteria given
+        /// </summary>
+        /// <param name="daysTillExpiry">The desire days till expiry from the current time</param>
+        /// <param name="strikeFromAtm">The desire strike price distance from the current underlying price</param>
+        /// <remarks>Applicable to Naked Call, Covered Call, and Protective Call Option Strategy</remarks>
+        /// <returns>Universe with filter applied</returns>
+        public OptionFilterUniverse SingleCall(int daysTillExpiry = 30, decimal strikeFromAtm = 0)
+        {
+            return SingleContract(OptionRight.Call, daysTillExpiry, strikeFromAtm);
+        }
+
+        /// <summary>
+        /// Sets universe of a single put contract with the closest match to criteria given
+        /// </summary>
+        /// <param name="daysTillExpiry">The desire days till expiry from the current time</param>
+        /// <param name="strikeFromAtm">The desire strike price distance from the current underlying price</param>
+        /// <remarks>Applicable to Naked Put, Covered Put, and Protective Put Option Strategy</remarks>
+        /// <returns>Universe with filter applied</returns>
+        public OptionFilterUniverse SinglePut(int daysTillExpiry = 30, decimal strikeFromAtm = 0)
+        {
+            return SingleContract(OptionRight.Put, daysTillExpiry, strikeFromAtm);
+        }
+
+        private OptionFilterUniverse SingleContract(OptionRight right, int daysTillExpiry = 30, decimal strikeFromAtm = 0)
+        {
+            // Select the expiry as the nearest to set days later
+            var expiry = AllSymbols.OrderBy(x => Math.Abs((x.ID.Date - _lastExchangeDate.AddDays(daysTillExpiry)).Days))
+                .First().ID.Date;
+            var contracts = AllSymbols.Where(x => x.ID.Date == expiry && x.ID.OptionRight == right);
+            // Select strike price
+            var selected = contracts.OrderBy(x => Math.Abs(x.ID.StrikePrice - Underlying.Price - strikeFromAtm)).First();
```

Comment:
I don't think the expiration filter here is behaving as expected, take the follow example

```
DaysTillExpiry 4
Today 8

Contract 1 expiration date 9
Contract 2 expiration date 15
Contract 3 expiration date 20

9 - (8 + 4) =>  Abs(-3) ? -> HERE
15 - (8 + 4) => Abs(3) -> HERE
20 - (8 + 4) => Abs(8)
```

We should have 1 centralized helper method location where we handle these daysTillExpiry/strikeFromAtm and reuse them in the whole file